libs/ledger-live-common/src/families/vechain/js-synchronisation.ts

```diff
@@ -0,0 +1,90 @@
+import type { GetAccountShape } from "../../bridge/jsHelpers";
+import { BigNumber } from "bignumber.js";
+import { makeSync, makeScanAccounts, mergeOps } from "../../bridge/jsHelpers";
+import { encodeAccountId, encodeTokenAccountId } from "../../account";
+import eip55 from "eip55";
+import { emptyHistoryCache } from "../../account";
+
+import { getAccount, getLastBlockHeight, getOperations, getTokenOperations } from "./api";
+import { findTokenById, getTokenById } from "@ledgerhq/cryptoassets/tokens";
+
+const getAccountShape: GetAccountShape = async info => {
+  const { initialAccount, currency, derivationMode } = info;
+  const address = eip55.encode(info.address);
+
+  const oldOperations = initialAccount?.operations || [];
+  const startAt = oldOperations.length ? (oldOperations[0].blockHeight || 0) + 1 : 1;
+
+  const accountId = encodeAccountId({
+    type: "js",
+    version: "2",
+    currencyId: currency.id,
+    xpubOrAddress: address,
+    derivationMode,
+  });
+
+  // get the current account balance state depending your api implementation
+  const { balance, energy } = await getAccount(address);
+
+  // get the current block height
+  const blockHeight = await getLastBlockHeight();
+
+  // Merge new operations with the previously synced ones
+  const newOperations = await getOperations(accountId, address, startAt);
+
+  //Get last token operations
+  const vthoAccountId = encodeTokenAccountId(accountId, findTokenById("vechain/vtho")!);
+  const vthoOperations = await getTokenOperations(
+    vthoAccountId,
+    address,
+    "0x0000000000000000000000000000456e65726779",
+    1,
```

Comment:
Maybe it would make sense to start at account's first operation block ? Ignore if it could cause problems

Reply:
Sorry I don't get what you mean

Reply:
I'll change it since  the first block seems to be 0, don't know why we set 1

Reply:
<img width="901" alt="Screenshot 2023-09-25 at 11 03 25" src="https://github.com/LedgerHQ/ledger-live/assets/5785838/782d2917-46fb-4d98-8859-546f1790232f">

Reply:
https://docs.vechain.org/thor/thorest-api